glean-core/src/storage/mod.rs

```diff
@@ -143,4 +143,41 @@ impl StorageManager {
 
         snapshot
     }
+
+    ///  Snapshot the experiments.
+    ///
+    /// ## Arguments:
+    ///
+    /// * `storage` - The database to get data from.
+    ///
+    /// ## Return value
+    ///
+    /// Returns a JSON representation of the stored data.
+    /// Returns `None` if no data for the store exists.
+    pub fn snapshot_experiments_as_json(
+        &self,
+        storage: &Database
+    ) -> Option<JsonValue> {
+        let mut snapshot: HashMap<String, HashMap<String, JsonValue>> = HashMap::new();
+
+        let mut snapshotter = |metric_name: &[u8], metric: &Metric| {
+            let metric_name = String::from_utf8_lossy(metric_name).into_owned();
+            if metric_name.ends_with("#experiment") {
+                let name = metric_name.splitn(2, '#').next().unwrap(); // safe unwrap, first field of a split always valid
+                let map = snapshot
+                    .entry(metric.category().into())
+                    .or_insert_with(HashMap::new);
+                map.insert(name.to_string(), metric.as_json());
+            }
+        };
+
+        // FIXME: glean_internal_info should probably not be known here or, at least, shared.
+        storage.iter_store_from(Lifetime::Application, "glean_internal_info".into(), &mut snapshotter);
```

Comment:
I'm probably missing a detail, but... if `glean_internal_info` *only* contains experiments, then would we need a special case snapshot function?  Could we just call `snapshot_as_json(glean, "glean_internal_info")`?

Reply:
`glean_internal_info` is the same storage used for sequence numbers and start_time, so it contains other things as well. However, they live in PingMaker

Reply:
Ah.  Would using a different store name *just* for experiments help?  i.e. `glean_experiments`?

Reply:
I guess so, but I think we should value consistency: other things do not have their own storage, we might want to think about this: do we want to transition other things (sequence numbers, start/end_time) to their own storage or keep them that way? What's the cleanest way going forward, in case we might want to add more things? Do we want an abstraction layer over this, e.g. "InternalMetricsDatabase" or something?